packages/dashboard/src/PanelManager.ts

```diff
@@ -208,35 +279,42 @@ class PanelManager {
     LayoutUtils.openComponent({ root, config, replaceConfig });
   }
 
-  handleDeleted(panelConfig) {
+  handleDeleted(panelConfig: ClosedPanel): void {
     log.debug2('Deleted:', panelConfig);
 
     this.removeClosedPanelConfig(panelConfig);
 
     this.sendUpdate();
   }
 
-  handleClosed(panelId, glContainer) {
+  handleClosed(panelId: string, glContainer: GoldenLayout.Container): void {
     // Panel component should be already unmounted at this point
     // so the emitted event sends the container object instead of the panel.
     log.debug2('Closed: ', panelId);
     this.addClosedPanel(glContainer);
     this.sendUpdate();
   }
 
-  addClosedPanel(glContainer) {
+  handleControlClose(id: string): void {
+    const config = { id };
+    const { root } = this.layout;
+    LayoutUtils.closeComponent(root, config);
+  }
+
+  addClosedPanel(glContainer: GoldenLayout.Container): void {
     const config = LayoutUtils.getComponentConfigFromContainer(glContainer);
-    if (config && this.dehydrateComponentConfigMap[config.component]) {
-      const dehydratedConfig = this.dehydrateComponentConfigMap[
-        config.component
-      ](config);
+    if (config) {
+      const dehydratedConfig = this.dehydrateComponent(
```

Comment:
Make sure this doesn't throw on unregistered components

Reply:
This path doesn't get triggered on unregistered components, since they don't have a `Panel` implemented that emits the CLOSED event.
With that said... it'd probably make sense to have the Panel wrapper right at the dashboard level, but I'll leave that as a possible future change.